### PR TITLE
[Playwright] Update CircleCI config for playwright workflows

### PR DESCRIPTION
**Pull request by aweng98**
Enable users with a shell cmd to run only DSM, or only DSS or all E2E tests on a PR branch on CircleCI. Allowed env are DEV and TEST.

Example cmd to run all DSS tests on DEV or TEST env: 
> `cd build-utils/`
> `./run_ci.sh run-e2e-tests dss playwright-circleci-config dev`
> `./run_ci.sh run-e2e-tests dss playwright-circleci-config test`

`run_ci.sh` --help:

Screenshot 2023-09-27 at 3 18 37 PM

### CODE REVIEW

File: playwright-e2e/utils/faker-utils.ts [hunk 1]
```diff
@@ -3,11 +3,11 @@ import { Page } from '@playwright/test';
 import * as user from 'data/fake-user.json';
 
 export const generateAlphaNumeric = (length?: number): string => {
-  return faker.random.alphaNumeric(length ? length : 6);
+  return faker.string.alphanumeric(length ? length : 6);
```
Comment (aweng98) [inline playwright-e2e/utils/faker-utils.ts:6]: `faker` deprecated functions.

File: playwright-e2e/utils/faker-utils.ts [hunk 2]
```diff
@@ -55,7 +55,6 @@ export const saveParticipantGuid = (guid: string) => {
  */
 export const simplifyShortID = (shortId: string, studyName: string): string => {
   const shortIdParts = shortId.split(`${studyName}_`); // Use 'RGP_' to determine where to split
-  const rgpPrefix = shortIdParts[0]; //RGP_ prefix
```
Comment (aweng98) [inline playwright-e2e/utils/faker-utils.ts:58]: Unused

File: .circleci/config.yml [hunk 3]
```diff
@@ -643,15 +643,15 @@ jobs:
             echo "export ENV=<< parameters.env >>" >> $BASH_ENV
             source $BASH_ENV
             .circleci/export-playwright-env.sh
-      - node/install-packages:
-          app-dir: *playwright_path
-          override-ci-command: npm install
```
Comment (aweng98) [inline .circleci/config.yml:648]: I didn't see this error before, this command is not working in `playwright-build` job.

Screenshot 2023-09-26 at 7 23 47 PM

File: .circleci/config.yml [hunk 4]
```diff
@@ -769,9 +779,6 @@ parameters:
   do-builds:
     type: boolean
     default: false
-  e2e-test-parallelism:
-    type: integer
-    default: 2
```
Comment (aweng98) [inline .circleci/config.yml:774]: Removed this pipeline param because no one ever used it.

File: .circleci/config.yml [hunk 5]
```diff
@@ -861,17 +868,19 @@ workflows:
...
   playwright-test-workflow:
     # Triggered by CI API (for example, run_ci.sh)
-    # Run Playwright tests against Dev and Test but not for Staging and Production
+    # Run tests supported on Dev or Test, but not for Staging and Production.
     when:
       and:
         - equal: [ "run-e2e-tests", << pipeline.parameters.api_call >> ]
+        - or:
+            - equal: [ "dev", << pipeline.parameters.deploy_env >> ]
+            - equal: [ "test", << pipeline.parameters.deploy_env >> ]
```
Comment (aweng98) [inline .circleci/config.yml:877]: Stops user from accidentally starting E2E tests on any higher env.

**Review by Rinchan20 — APPROVED** [review]

File: .circleci/config.yml [hunk 6]
```diff
@@ -670,12 +670,12 @@ jobs:
...
-        default: 1
+        default: 2
       test_suite:
-        # to trigger E2E tests with labels matching study key:
-        # [basil, osteo, angio, brain, mbc, testboston, mpc, atcp, rarex, rgp, esc, prion, cgc, circadia, pancan, singular, brugada, lms, fon]
-        type: string
-        default: << pipeline.parameters.study_key >>
+        # Trigger DSS, DSM or all tests
+        type: enum
+        enum: [ dss, dsm, UNKNOWN ]
```
Comment (ssettipalli) [inline .circleci/config.yml:677]: Does UNKNOWN trigger both dss & dsm ?

File: build-utils/run_ci.sh [hunk 7]
```diff
@@ -5,7 +5,6 @@ COMMAND=$1
 STUDY_KEY=$2
 BRANCH=$3
 TARGET_ENV=$4
-E2E_TEST_PARALLELISM=${5:-2}
```
Comment (ssettipalli) [inline build-utils/run_ci.sh:8]: No change to build and deploy of different studies running parallel, right ? 
Ex: core-sdk change / merge triggering All studies deploy in parallel.

Comment (aweng98) [reply build-utils/run_ci.sh:8]: no change to other like deploy jobs. only on playwright job.

File: .circleci/config.yml (hunk 6 above)
Comment (aweng98) [reply .circleci/config.yml:677]: yes, that would be all tests.

**Pull request merged**